ospfd/ospf_vty.c

```diff
@@ -10064,27 +10064,42 @@ static int config_write_ospf_area(struct vty *vty, struct ospf *ospf)
 		for (rn1 = route_top(area->ranges); rn1; rn1 = route_next(rn1))
 			if (rn1->info) {
 				struct ospf_area_range *range = rn1->info;
-
-				vty_out(vty, " area %s range %s/%d", buf,
-					inet_ntoa(rn1->p.u.prefix4),
-					rn1->p.prefixlen);
+				int output_default = 1;
 
 				if (range->cost_config
-				    != OSPF_AREA_RANGE_COST_UNSPEC)
-					vty_out(vty, " cost %d",
+				    != OSPF_AREA_RANGE_COST_UNSPEC) {
+					vty_out(vty," area %s range %s/%d"
+						" cost %d\n", buf,
+						inet_ntoa(rn1->p.u.prefix4),
+						rn1->p.prefixlen,
 						range->cost_config);
+					output_default = 0;
+				}
 
 				if (!CHECK_FLAG(range->flags,
-						OSPF_AREA_RANGE_ADVERTISE))
-					vty_out(vty, " not-advertise");
-
+						OSPF_AREA_RANGE_ADVERTISE)) {
+					vty_out(vty," area %s range %s/%d"
+						" not-advertise\n", buf,
```

Comment:
Please don't break user visible strings. You are free to exceed the 80 column limit in this case.

Reply:
Yeah. I think you're right here. The issue is the fact that we allow the combination al all.
so configuring "cost" then "area" combines them, but deleting either removes both. I suspect the "correct" approach would be to make the latest range command take preference and not do the combination. i.e.cost removes not-advertise and vice-versa. 
does that seem like a better approach. The current situation confuses frr-reload as it cannot delete the combined line.

Reply:
Looks like I tested with an old master - I had this
pruddy-Standard-PC-i440FX-PIIX-1996(config-router)# area 0 range 10.0.0.0/24 cost 100
pruddy-Standard-PC-i440FX-PIIX-1996(config-router)# area 0 range 10.0.0.0/24 not-advertise 
pruddy-Standard-PC-i440FX-PIIX-1996(config-router)# do sh run
Building configuration...

Current configuration:
!
frr version 5.1-dev-MyOwnFRRVersion
frr defaults traditional
hostname pruddy-Standard-PC-i440FX-PIIX-1996
log syslog informational
no ip forwarding
no ipv6 forwarding
!
router ospf
 area 0 range 10.0.0.0/24 cost 100 not-advertise

but updating to latest fixes it - apologies for the noise.